### ISSUE TITLE
Missing warning for unreachable arm in the presence of an intermediate when clause

### ISSUE THREAD

**Issue by alrz** [dotnet/fsharp#14573]
Please provide a succinct description of the issue.

**Repro steps**

```fs
let f value b = match value  with
                | (true, true) -> -1
                | _ when b -> 2
                | (true, false) -> 3
                | (true, _) -> 4 // expected warning
```

**Expected behavior**

`warning FS0026: This rule will never be matched` for the last arm

**Actual behavior**

No warning about unreachable arm

**Comment by vzarytovskii** [comment]
I can see the warning on the 7.0.100+ FSI, @alrz which compiler/fsi version are you using?

**Comment by alrz** [comment]
Thanks. I'm using 6.0.1 fwiw.

**Comment by vzarytovskii** [comment]
> Thanks. I'm using FSharp.Core 6.0.1 FWIW.

I will check what was the fix and whether it was backported to 6.0.x

**Comment by alrz** [comment]
That would actually help, I'm trying out the same clause-by-clause approach in C# for "problematic" when clauses and sounds like I managed to port an old bug. Could it be https://github.com/dotnet/fsharp/pull/11993? Because that does change the detection method.
